mdoc 5.5 fails to reflect UWP Community Toolkit documentation

Binaries attached:
[uwp.zip](https://github.com/mono/api-doc-tools/files/1741821/uwp.zip)

This is also dependent on UWP assemblies, referenced in my script via:

```
 -L "C:\Program Files (x86)\Windows Kits\10\References\10.0.15063.0"
```

Error:

```
2018-02-20T20:50:39.4288283Z Member Added: public void StopEnumeration ();
2018-02-20T20:50:39.4444551Z Member Added: public static Microsoft.Toolkit.Uwp.Connectivity.BluetoothLEHelper Context { get; }
2018-02-20T20:50:39.4444551Z Member Added: public static bool IsBluetoothLESupported { get; }
2018-02-20T20:50:39.5381822Z ##[error]mdoc: System.NullReferenceException: Object reference not set to an instance of an object.
   at Mono.Documentation.Updater.Formatters.CppFormatters.CppFullMemberFormatter.AppendTypeName(StringBuilder buf, TypeReference type, DynamicParserContext context)
   at Mono.Documentation.Updater.Formatters.CppFormatters.CppFullMemberFormatter.AppendGenericType(StringBuilder buf, TypeReference type, DynamicParserContext context, Boolean appendGeneric)
   at Mono.Documentation.Updater.MemberFormatter._AppendTypeName(StringBuilder buf, TypeReference type, DynamicParserContext context, Boolean appendGeneric)
   at Mono.Documentation.Updater.MemberFormatter.GetTypeName(TypeReference type, DynamicParserContext context, Boolean appendGeneric)
   at Mono.Documentation.Updater.Formatters.CppFormatters.CppFullMemberFormatter.GetTypeNameWithOptions(TypeReference type, Boolean appendHat, Boolean appendGeneric)
   at Mono.Documentation.Updater.Formatters.CppFormatters.CppFullMemberFormatter.GetPropertyDeclaration(PropertyDefinition property)
   at Mono.Documentation.Updater.MemberFormatter.GetDeclaration(IMemberDefinition member)
   at Mono.Documentation.Updater.MemberFormatter.GetDeclaration(MemberReference mreference)
   at Mono.Documentation.MDocUpdater.UpdateSignature(MemberFormatter formatter, MemberReference member, XmlElement xmlElement)
   at Mono.Documentation.MDocUpdater.UpdateMember(DocsNodeInfo info, FrameworkTypeEntry typeEntry, Dictionary`2 implementedMembers)
   at Mono.Documentation.MDocUpdater.MakeMember(XmlDocument doc, DocsNodeInfo info, XmlNode members, FrameworkTypeEntry typeEntry, Dictionary`2 iplementedMembers)
   at Mono.Documentation.MDocUpdater.DoUpdateType2(String message, XmlDocument basefile, TypeDefinition type, FrameworkTypeEntry typeEntry, String output, Boolean insertSince)
   at Mono.Documentation.MDocUpdater.StubType(TypeDefinition type, String output, IEnumerable`1 importers, String Id, String Version)
   at Mono.Documentation.MDocUpdater.DoUpdateType(TypeDefinition type, FrameworkTypeEntry typeEntry, String basepath, String dest)
   at Mono.Documentation.MDocUpdater.DoUpdateAssembly(AssemblySet assemblySet, AssemblyDefinition assembly, XmlElement index_types, String source, String dest, HashSet`1 goodfiles)
   at Mono.Documentation.MDocUpdater.DoUpdateAssemblies(String source, String dest)
   at Mono.Documentation.MDocUpdater.Run(IEnumerable`1 args)
   at Mono.Documentation.MDoc.Run(String[] args)
   at Mono.Documentation.MDoc.Main(String[] args)
See `mdoc help' for more information or use --debug to diagnose.
```

--- a/mdoc/Consts.cs
+++ b/mdoc/Consts.cs
@@ -3,7 +3,7 @@ namespace Mono.Documentation
 {
 	public static class Consts
 	{
-		public static string MonoVersion = "5.5.0.0";
+		public static string MonoVersion = "5.5.0.1";
 		public const string DocId = "DocId";
 		public const string CppCli = "C++ CLI";
 	    public const string CppCx = "C++ CX";

--- a/mdoc/Mono.Documentation/MDocUpdater.cs
+++ b/mdoc/Mono.Documentation/MDocUpdater.cs
@@ -506,12 +506,12 @@ namespace Mono.Documentation
             Message (TraceLevel.Warning, "mdoc: " + format, args);
         }
 
-        internal AssemblyDefinition LoadAssembly (string name, IAssemblyResolver assemblyResolver)
+        internal AssemblyDefinition LoadAssembly (string name, IMetadataResolver resolver,  IAssemblyResolver assemblyResolver)
         {
             AssemblyDefinition assembly = null;
             try
             {
-                assembly = AssemblyDefinition.ReadAssembly (name, new ReaderParameters { AssemblyResolver = assemblyResolver });
+                assembly = AssemblyDefinition.ReadAssembly (name, new ReaderParameters { AssemblyResolver = assemblyResolver, MetadataResolver = resolver });
             }
             catch (Exception ex)
             {

--- a/mdoc/Mono.Documentation/Updater/Formatters/CppFormatters/CppCxFullMemberFormatter.cs
+++ b/mdoc/Mono.Documentation/Updater/Formatters/CppFormatters/CppCxFullMemberFormatter.cs
@@ -508,7 +508,11 @@ namespace Mono.Documentation.Updater.Formatters.CppFormatters
             try
             {
                 var typedef = fdef.FieldType.Resolve();
-                isEnumFieldType = typedef.IsEnum;
+
+                if (typedef != null)
+                    isEnumFieldType = typedef.IsEnum;
+                else
+                    isEnumFieldType = false;
             }
             catch
             {

--- a/mdoc/Mono.Documentation/Updater/Formatters/CppFormatters/CppFullMemberFormatter.cs
+++ b/mdoc/Mono.Documentation/Updater/Formatters/CppFormatters/CppFullMemberFormatter.cs
@@ -106,6 +106,9 @@ namespace Mono.Documentation.Updater.Formatters.CppFormatters
         protected override StringBuilder AppendTypeName (StringBuilder buf, TypeReference type, DynamicParserContext context)
         {
             string typeFullName = type.FullName;
+            if (string.IsNullOrWhiteSpace (typeFullName))
+                return buf;
+            
             if (!typeFullName.StartsWith ("System."))
             {
                 return base.AppendTypeName(buf, type, context);

--- a/mdoc/Mono.Documentation/Updater/Formatters/CppFormatters/CppFullMemberFormatter.cs
+++ b/mdoc/Mono.Documentation/Updater/Formatters/CppFormatters/CppFullMemberFormatter.cs
@@ -547,7 +547,7 @@ namespace Mono.Documentation.Updater.Formatters.CppFormatters
                 TypeReference declDef;
                 try
                 {
-                    declDef = decl.Resolve(); 
+                    declDef = decl.Resolve() ?? decl; 
                 }
                 catch
                 {

--- a/mdoc/Mono.Documentation/Updater/Frameworks/AssemblySet.cs
+++ b/mdoc/Mono.Documentation/Updater/Frameworks/AssemblySet.cs
@@ -14,6 +14,8 @@ namespace Mono.Documentation.Updater.Frameworks
     {
         readonly BaseAssemblyResolver resolver = new Frameworks.MDocResolver ();
         IAssemblyResolver cachedResolver;
+        IMetadataResolver metadataResolver;
+
         HashSet<string> assemblyPaths = new HashSet<string> ();
         Dictionary<string, bool> assemblyPathsMap = new Dictionary<string, bool> ();
         HashSet<string> assemblySearchPaths = new HashSet<string> ();
@@ -26,6 +28,7 @@ namespace Mono.Documentation.Updater.Frameworks
         public AssemblySet (string name, IEnumerable<string> paths, IEnumerable<string> resolverSearchPaths, IEnumerable<string> imports = null, string version = null, string id = null)
         {
             this.cachedResolver = new CachedResolver (this.resolver);
+            this.metadataResolver = new Frameworks.MDocMetadataResolver (this.cachedResolver);
 
             Name = name;
             Version = version;
@@ -127,7 +130,7 @@ namespace Mono.Documentation.Updater.Frameworks
 		IEnumerable<AssemblyDefinition> LoadAllAssemblies ()
 		{
 			foreach (var path in this.assemblyPaths) {
-                var assembly = MDocUpdater.Instance.LoadAssembly (path, this.cachedResolver);
+                var assembly = MDocUpdater.Instance.LoadAssembly (path, this.metadataResolver, this.cachedResolver);
 				if (assembly != null) {
 					foreach (var type in assembly.MainModule.ExportedTypes.Where (t => t.IsForwarder).Select (t => t.FullName))
 						forwardedTypes.Add (type);

--- a/mdoc/Mono.Documentation/Updater/Frameworks/FrameworkTypeEntry.cs
+++ b/mdoc/Mono.Documentation/Updater/Frameworks/FrameworkTypeEntry.cs
@@ -48,7 +48,7 @@ namespace Mono.Documentation.Updater.Frameworks
 			try {
                 var sig = formatter.GetDeclaration (member);
 				memberscsharpsig.Add(sig);
-                if (!sigMap.ContainsKey (sig))
+                if (sig != null && !sigMap.ContainsKey (sig))
                     sigMap.Add (sig, true);
 			}
 			catch {}

--- a/mdoc/Mono.Documentation/Updater/Frameworks/MDocMetadataResolver.cs
+++ b/mdoc/Mono.Documentation/Updater/Frameworks/MDocMetadataResolver.cs
@@ -1,0 +1,111 @@
+﻿using System;
+using Mono.Cecil;
+
+namespace Mono.Documentation.Updater.Frameworks
+{
+    class MDocMetadataResolver : MetadataResolver
+    {
+        public MDocMetadataResolver (IAssemblyResolver r) : base (r) { }
+
+        ReaderParameters parameters = new ReaderParameters ();
+
+        public override TypeDefinition Resolve (TypeReference type)
+        {
+            type = type.GetElementType ();
+
+            var scope = type.Scope;
+
+            if (scope == null)
+                return null;
+            
+            CachedResolver assembly_resolver = this.AssemblyResolver as CachedResolver;
+            parameters.AssemblyResolver = assembly_resolver;
+            parameters.MetadataResolver = this;
+
+            switch (scope.MetadataScopeType)
+            {
+                case MetadataScopeType.AssemblyNameReference:
+                    var assembly = assembly_resolver.ResolveCore ((AssemblyNameReference)scope, parameters, type);
+                    if (assembly == null)
+                        return null;
+
+                    return GetType (assembly.MainModule, type);
+                case MetadataScopeType.ModuleDefinition:
+                    return GetType ((ModuleDefinition)scope, type);
+                case MetadataScopeType.ModuleReference:
+                    var modules = type.Module.Assembly.Modules;
+                    var module_ref = (ModuleReference)scope;
+                    for (int i = 0; i < modules.Count; i++)
+                    {
+                        var netmodule = modules[i];
+                        if (netmodule.Name == module_ref.Name)
+                            return GetType (netmodule, type);
+                    }
+                    break;
+            }
+
+            throw new NotSupportedException ();
+        }
+        static TypeDefinition GetType (ModuleDefinition module, TypeReference reference)
+        {
+            var type = GetTypeDefinition (module, reference);
+            if (type != null)
+                return type;
+
+            if (!module.HasExportedTypes)
+                return null;
+
+            var exported_types = module.ExportedTypes;
+
+            for (int i = 0; i < exported_types.Count; i++)
+            {
+                var exported_type = exported_types[i];
+                if (exported_type.Name != reference.Name)
+                    continue;
+
+                if (exported_type.Namespace != reference.Namespace)
+                    continue;
+
+                return exported_type.Resolve ();
+            }
+
+            return null;
+        }
+
+        static TypeDefinition GetTypeDefinition (ModuleDefinition module, TypeReference type)
+        {
+            if (!type.IsNested)
+                return module.GetType (type.Namespace, type.Name);
+
+            var declaring_type = type.DeclaringType.Resolve ();
+            if (declaring_type == null)
+                return null;
+
+            return GetNestedType (declaring_type, TypeFullName(type));
+        }
+
+        public static TypeDefinition GetNestedType (TypeDefinition self, string fullname)
+        {
+            if (!self.HasNestedTypes)
+                return null;
+
+            var nested_types = self.NestedTypes;
+
+            for (int i = 0; i < nested_types.Count; i++)
+            {
+                var nested_type = nested_types[i];
+
+                if (TypeFullName(nested_type) == fullname)
+                    return nested_type;
+            }
+
+            return null;
+        }
+        public static string TypeFullName (TypeReference self)
+        {
+            return string.IsNullOrEmpty (self.Namespace)
+                ? self.Name
+                : self.Namespace + '.' + self.Name;
+        }
+    }
+}

--- a/mdoc/Mono.Documentation/Updater/Frameworks/MDocResolver.cs
+++ b/mdoc/Mono.Documentation/Updater/Frameworks/MDocResolver.cs
@@ -22,18 +22,47 @@ namespace Mono.Documentation.Updater.Frameworks
     {
         public override AssemblyDefinition Resolve (AssemblyNameReference name, ReaderParameters parameters)
         {
+            return Resolve (name, parameters, null, null);
+        }
+
+        internal AssemblyDefinition Resolve (AssemblyNameReference name, ReaderParameters r, TypeReference forType, List<string> exportedFiles)
+        {
+            if (exportedFiles == null)
+                exportedFiles = new List<string> ();
+
+            var a = this.ResolveCore (name, r, exportedFiles);
+            //Console.WriteLine($"resolver, resolving {forType.FullName} in {name}");
+
+
+            if (forType != null && a.MainModule.HasExportedTypes) {
+                var etype = a.MainModule.ExportedTypes.SingleOrDefault (t => t.FullName == forType.FullName) as ExportedType;
+                if (etype != null)
+                {
+                    string file = a.MainModule.FileName;
+                    AssemblyNameReference exportedTo = (AssemblyNameReference)etype.Scope;
+                    Console.WriteLine ($"resolving {forType.FullName} in {name.FullName}. Found {file}, but it's exported to {exportedTo.FullName}");
+                    exportedFiles.Add (file);
+                    return Resolve (exportedTo, r, forType, exportedFiles);
+                }
+            }
+
+            return a;
+        }
+
+        AssemblyDefinition ResolveCore (AssemblyNameReference name, ReaderParameters parameters, IEnumerable<string> assembliesToIgnore)
+        {
             var ver = name.Version;
             if (ver.Major == 255 && ver.Minor == 255 && ver.Revision == 255 && name.Name == "mscorlib")
             {
                 var v = new Version (4, 5, 0);
                 var anr = new AssemblyNameReference (name.Name, v);
-                return base.Resolve (anr, parameters);
+                return base.Resolve (anr, parameters, assembliesToIgnore);
             }
             else
-                return base.Resolve (name, parameters);
+                return base.Resolve (name, parameters, assembliesToIgnore);
         }
 
-        IEnumerable<AssemblyDefinition> GetInstalledAssemblies (AssemblyNameReference name, ReaderParameters parameters)
+        IEnumerable<AssemblyDefinition> GetInstalledAssemblies (AssemblyNameReference name, ReaderParameters parameters, IEnumerable<string> filesToIgnore)
         {
             AssemblyDefinition assembly;
             if (name.IsRetargetable)
@@ -49,7 +78,7 @@ namespace Mono.Documentation.Updater.Frameworks
 
             if (IsZero(name.Version))
             {
-                assembly = base.SearchDirectory(name, framework_dirs, parameters);
+                assembly = base.SearchDirectory(name, framework_dirs, parameters, filesToIgnore);
                 if (assembly != null)
                     yield return assembly;
             }
@@ -65,12 +94,12 @@ namespace Mono.Documentation.Updater.Frameworks
             if (assembly != null)
                 yield return assembly;
 
-            assembly = base.SearchDirectory(name, framework_dirs, parameters);
+            assembly = base.SearchDirectory(name, framework_dirs, parameters, filesToIgnore);
             if (assembly != null)
                 yield return assembly;
         }
 
-        protected override AssemblyDefinition SearchDirectory(AssemblyNameReference name, IEnumerable<string> directories, ReaderParameters parameters)
+        protected override AssemblyDefinition SearchDirectory(AssemblyNameReference name, IEnumerable<string> directories, ReaderParameters parameters, IEnumerable<string> filesToIgnore)
         {
             // look for an assembly that matches the name in all the search directories
             string[] extensions = new[] { ".dll", ".exe", ".winmd" };
@@ -79,7 +108,7 @@ namespace Mono.Documentation.Updater.Frameworks
                 .SelectMany(d => extensions.Select(e => Path.Combine(d, name.Name + e)))
                 .Distinct();
             var namedPaths = npaths
-                .Where(f => File.Exists(f));
+                .Where(f => File.Exists(f) && !filesToIgnore.Any (fi => fi != f));
 
             if (!namedPaths.Any()) return null;
 
@@ -94,7 +123,7 @@ namespace Mono.Documentation.Updater.Frameworks
             
             var applicableVersions = namedPaths
                 .Select (getAssemblies)
-                .Concat (GetInstalledAssemblies(name, parameters))
+                .Concat (GetInstalledAssemblies(name, parameters, filesToIgnore))
                 .Select (a => new
                 {
                     Assembly = a,
@@ -187,8 +216,15 @@ namespace Mono.Documentation.Updater.Frameworks
         Dictionary<string, int> dict = new Dictionary<string, int> ();
         Queue<float> traces = new Queue<float> (10);
 
-        public AssemblyDefinition Resolve (AssemblyNameReference name, ReaderParameters parameters)
+        public AssemblyDefinition Resolve (AssemblyNameReference name, ReaderParameters parameters) 
         {
+            return ResolveCore (name, parameters, null);
+        }
+
+        internal AssemblyDefinition ResolveCore (AssemblyNameReference name, ReaderParameters parameters, TypeReference forType)
+        {
+            string cacheKey = name.FullName;
+
             // book keeping to detect stackoverflow conditions 
             //Console.WriteLine ($"resolving: {name.Name} {name.Version} - {Environment.StackTrace.Length}");
             float sum = (float)traces.Sum ();
@@ -221,7 +257,7 @@ namespace Mono.Documentation.Updater.Frameworks
                 Console.WriteLine ("Possible StackOverFlow condition detected. The following assemblies are being resolved");
                 foreach(var item in dict) {
                     
-                    if (cache.TryGetValue (name.FullName, out assembly)) {
+                    if (cache.TryGetValue (item.Key, out assembly)) {
                         Console.WriteLine ($"[] r {item.Key}{Environment.NewLine}   l {assembly.FullName}{Environment.NewLine}   #{item.Value} times.");    
                     }
                     else {
@@ -231,11 +267,27 @@ namespace Mono.Documentation.Updater.Frameworks
                 Console.WriteLine ("Will attempt to continue;");
             }
 
-            if (cache.TryGetValue (name.FullName, out assembly))
+            if (cache.TryGetValue (cacheKey, out assembly))
+            {
+                if (forType != null && assembly.MainModule.HasExportedTypes) {
+                    // check to see if the type is in the exported assemblies
+                    var etype = assembly.MainModule.ExportedTypes.SingleOrDefault (et => et.FullName == forType.FullName);
+                    if (etype != null) {
+                        var escope = (AssemblyNameReference)etype.Scope;
+                        if (cache.TryGetValue (escope.FullName, out assembly)) {
+                            return assembly;
+                        }
+                        else {
+                            assembly = ((MDocResolver)this.Resolver).Resolve (escope, parameters, forType, null);
+                            cache[escope.FullName] = assembly;
+                        }
+                    }
+                }
                 return assembly;
+            }
 
-            assembly = this.Resolver.Resolve (name, parameters);
-            cache[name.FullName] = assembly;
+            assembly = ((MDocResolver)this.Resolver).Resolve (name, parameters, forType, null);
+            cache[cacheKey] = assembly;
 
             return assembly;
         }
@@ -294,10 +346,17 @@ namespace Mono.Documentation.Updater.Frameworks
             return this.Resolve (name, new ReaderParameters ());
         }
 
+        string[] emptyStringArray = new string[0];
+
         public override AssemblyDefinition Resolve (AssemblyNameReference name, ReaderParameters parameters)
         {
+            return Resolve (name, parameters, emptyStringArray);
+        }
+
+        internal AssemblyDefinition Resolve (AssemblyNameReference name, ReaderParameters parameters, IEnumerable<string> filesToIgnore)
+        {
             var directories = this.GetSearchDirectories ();
-            var assembly = SearchDirectory (name, directories, parameters);
+            var assembly = SearchDirectory (name, directories, parameters, filesToIgnore);
             if (assembly != null)
                 return assembly;
 
@@ -314,7 +373,7 @@ namespace Mono.Documentation.Updater.Frameworks
 
             if (IsZero (name.Version))
             {
-                assembly = SearchDirectory (name, framework_dirs, parameters);
+                assembly = SearchDirectory (name, framework_dirs, parameters, filesToIgnore);
                 if (assembly != null)
                     return assembly;
             }
@@ -330,14 +389,14 @@ namespace Mono.Documentation.Updater.Frameworks
             if (assembly != null)
                 return assembly;
 
-            assembly = SearchDirectory (name, framework_dirs, parameters);
+            assembly = SearchDirectory (name, framework_dirs, parameters, filesToIgnore);
             if (assembly != null)
                 return assembly;
 
             throw new AssemblyResolutionException (name);
         }
 
-        protected virtual AssemblyDefinition SearchDirectory (AssemblyNameReference name, IEnumerable<string> directories, ReaderParameters parameters)
+        protected virtual AssemblyDefinition SearchDirectory (AssemblyNameReference name, IEnumerable<string> directories, ReaderParameters parameters, IEnumerable<string> filesToIgnore)
         {
             var extensions = name.IsWindowsRuntime ? new[] { ".winmd", ".dll" } : new[] { ".exe", ".dll" };
             foreach (var directory in directories)
@@ -345,7 +404,7 @@ namespace Mono.Documentation.Updater.Frameworks
                 foreach (var extension in extensions)
                 {
                     string file = Path.Combine (directory, name.Name + extension);
-                    if (!File.Exists (file))
+                    if (!File.Exists (file) || filesToIgnore.Any (f => f == file))
                         continue;
                     try
                     {

--- a/mdoc/Mono.Documentation/Updater/Frameworks/MDocResolver.cs
+++ b/mdoc/Mono.Documentation/Updater/Frameworks/MDocResolver.cs
@@ -20,9 +20,6 @@ namespace Mono.Documentation.Updater.Frameworks
     /// a UWP library. </para></remarks>
     class MDocResolver : MDocBaseResolver
     {
-        IEnumerable<string> fxpaths;
-        IEnumerable<string> gacpaths;
-
         public override AssemblyDefinition Resolve (AssemblyNameReference name, ReaderParameters parameters)
         {
             var ver = name.Version;
@@ -185,11 +182,55 @@ namespace Mono.Documentation.Updater.Frameworks
             return Resolve (name, new ReaderParameters () { AssemblyResolver = this });
         }
 
+        float lastAverageStackLength = 0;
+        int stackTraceIncrease = 0;
+        Dictionary<string, int> dict = new Dictionary<string, int> ();
+        Queue<float> traces = new Queue<float> (10);
 
         public AssemblyDefinition Resolve (AssemblyNameReference name, ReaderParameters parameters)
         {
+            // book keeping to detect stackoverflow conditions 
+            //Console.WriteLine ($"resolving: {name.Name} {name.Version} - {Environment.StackTrace.Length}");
+            float sum = (float)traces.Sum ();
+            float count = (float)traces.Count;
+            float averageStackLength = sum / count;
+            if (float.IsNaN (averageStackLength)) {
+                averageStackLength = Environment.StackTrace.Length;
+            }
+            //Console.WriteLine (averageStackLength);
+            if (lastAverageStackLength < averageStackLength) {
+                
+                if (!dict.ContainsKey (name.FullName))
+                    dict.Add (name.FullName, 0);
+                dict[name.FullName]++;
+                stackTraceIncrease++;
+            }
+            else {
+                stackTraceIncrease = 0;
+                dict.Clear ();
+            }
+
+            if (traces.Count > 10)
+                traces.Dequeue ();
+            traces.Enqueue (Environment.StackTrace.Length);
+            lastAverageStackLength = averageStackLength;
 
             AssemblyDefinition assembly;
+
+            if (stackTraceIncrease > 50)  {
+                Console.WriteLine ("Possible StackOverFlow condition detected. The following assemblies are being resolved");
+                foreach(var item in dict) {
+                    
+                    if (cache.TryGetValue (name.FullName, out assembly)) {
+                        Console.WriteLine ($"[] r {item.Key}{Environment.NewLine}   l {assembly.FullName}{Environment.NewLine}   #{item.Value} times.");    
+                    }
+                    else {
+                        Console.WriteLine ($"[] resolving {item.Key}, #{item.Value} times.");    
+                    }
+                }
+                Console.WriteLine ("Will attempt to continue;");
+            }
+
             if (cache.TryGetValue (name.FullName, out assembly))
                 return assembly;
 

--- a/mdoc/Mono.Documentation/Updater/Frameworks/MDocResolver.cs
+++ b/mdoc/Mono.Documentation/Updater/Frameworks/MDocResolver.cs
@@ -108,7 +108,7 @@ namespace Mono.Documentation.Updater.Frameworks
                 .SelectMany(d => extensions.Select(e => Path.Combine(d, name.Name + e)))
                 .Distinct();
             var namedPaths = npaths
-                .Where(f => File.Exists(f) && !filesToIgnore.Any (fi => fi != f));
+                .Where(f => File.Exists(f) && !filesToIgnore.Any (fi => fi == f));
 
             if (!namedPaths.Any()) return null;
 

--- a/mdoc/mdoc.csproj
+++ b/mdoc/mdoc.csproj
@@ -132,6 +132,7 @@
     <Compile Include="Mono.Documentation\Updater\Frameworks\AssemblySet.cs" />
     <Compile Include="Mono.Documentation\Updater\Frameworks\MDocResolver.cs" />
     <Compile Include="Mono.Documentation\MDocUpdater.cs" />
+    <Compile Include="Mono.Documentation\Updater\Frameworks\MDocMetadataResolver.cs" />
   </ItemGroup>
   <ItemGroup>
     <EmbeddedResource Include="Resources\defaulttemplate.xsl">

--- a/mdoc/mdoc.nuspec
+++ b/mdoc/mdoc.nuspec
@@ -2,7 +2,7 @@
 <package >
   <metadata>
     <id>mdoc</id>
-    <version>5.5.0.0</version>
+    <version>5.5.0.1</version>
     <title>mdoc</title>
     <authors>Microsoft</authors>
     <owners>Microsoft</owners>

